app/models/web_authenticator.rb

```diff
@@ -1,6 +1,14 @@
-class WebAuthenticator < Struct.new(:controller)
+class WebAuthenticator < Struct.new(:controller, :params)
   def current_user
-    @current_user ||= User.where(id: controller.session[:user_id]).first || Guest.new
+    if @current_user.nil?
+      @current_user = if params[:token]
+                        GuestWithToken.new(params[:token])
```

Comment:
I think that's fine. that is also what happens when trying to visit a published auction via regular `id`.

This is in line with the philosophy of "exceptions should be exceptional" -- I don't have a good blog post ref on that one, but the idea is that you want exceptions to raise most of the time because that means something has gone wrong and only by raising will you see it in the dev logs (perhaps wrong link in an email or something)